prow/tide/tide.go

```diff
@@ -1731,6 +1750,8 @@ func headContexts(log *logrus.Entry, ghc githubClient, pr *PullRequest) ([]Conte
 	org := string(pr.Repository.Owner.Login)
 	repo := string(pr.Repository.Name)
 	// Log this event so we can tune the number of commits we list to minimize this.
+	// TODO alvaroaleman: Add checkrun support here. Doesn't seem to happen often though,
+	// openshift doesn't have a single occurrence of this in the past seven days.
```

Comment:
~The failure mode for this is pretty bad if the checkruns are not required for merge by branch protection. We'd merge the PR without passing tests. In particular someone could intentionally bypass tests by pushing a large number of PR commits.~ Looks like we would instead filter the PR out of the pool which isn't ideal, but isn't dangerous.

Reply:
This is a property of a PR and something leaking from the backend - if it happens for a PR once, it will happen every time, we usually notice. We do see this from time to time and detecting it will be really tough - the first time someone has to figure out why a PR is not merging when it's got failing check runs and hitting this API edge case, they will likely spend an enormous amount of time debugging. We should consider doing this in the current PR or a quick follow-up, I do not want to be the clueless `tide` admin who gets the question about this.

Reply:
Yeah, i'll be doing a follow-up soon for this